stablehlo/dialect/TypeInference.h

```diff
@@ -0,0 +1,150 @@
+/* Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+   Copyright 2022 The StableHLO Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#ifndef STABLEHLO_DIALECT_STABLEHLO_TYPE_INFERENCE_H
+#define STABLEHLO_DIALECT_STABLEHLO_TYPE_INFERENCE_H
+
+#include "mlir/IR/BuiltinAttributes.h"
+#include "mlir/IR/BuiltinTypes.h"
+#include "mlir/IR/Location.h"
+#include "mlir/IR/Types.h"
+#include "mlir/Interfaces/InferTypeOpInterface.h"
+#include "mlir/Support/LogicalResult.h"
+
+namespace mlir {
+namespace hlo {
+
```

Comment:
Let's add a comment similarly to the comment for shape functions:

```
//===----------------------------------------------------------------------===//
// Utilities for shape functions
//===----------------------------------------------------------------------===//
// TODO(#270): Remove them when all shape functions are moved to this file.
```

Reply:
Done.